TransferMethod/Sources/View/Widget/TextWidget.swift

```diff
@@ -69,6 +89,183 @@ class TextWidget: AbstractWidget {
+            return finalText
+        }
+        return inputText
+    }
+
+    // swiftlint:disable function_parameter_count
+    private func applyFormatForPatternCharacter(currentPatternCharacter: [Character],
+                                                currentText: String,
+                                                currentTextCharacter: String,
+                                                currentTextIndex: inout String.Index,
+                                                finalText: inout String,
+                                                isEscapedCharacter: inout Bool,
+                                                pattern: String,
+                                                patternIndex: inout String.Index,
+                                                patternCharactersToBeWritten: inout String) {
+        switch currentPatternCharacter.first {
+        case PatternCharacter.lettersAndNumbersPatternCharacter.rawValue:
+            handleTextForLettersAndNumbers(currentText: currentText,
+                                           currentTextCharacter: currentTextCharacter,
+                                           currentTextIndex: &currentTextIndex,
+                                           finalText: &finalText,
+                                           pattern: pattern,
+                                           patternCharactersToBeWritten: &patternCharactersToBeWritten,
+                                           patternIndex: &patternIndex)
+
+        case PatternCharacter.lettersOnlyPatternCharacter.rawValue,
+             PatternCharacter.numbersOnlyPatternCharacter.rawValue:
+            let filteredCharacter =
+                getTextForPatternCharacter(currentPatternCharacter.first!, currentTextCharacter)
+            handleTextForLettersAndNumbers(currentText: currentText,
+                                           currentTextCharacter: filteredCharacter,
+                                           currentTextIndex: &currentTextIndex,
+                                           finalText: &finalText,
+                                           pattern: pattern,
+                                           patternCharactersToBeWritten: &patternCharactersToBeWritten,
+                                           patternIndex: &patternIndex)
+
+        default:
+            isEscapedCharacter = self.isEscapedCharacter(currentPatternCharacter.first!)
+            if !isEscapedCharacter {
+                if String(currentPatternCharacter) == currentTextCharacter {
+                    finalText += currentTextCharacter
+                    currentTextIndex = currentText.index(after: currentTextIndex)
+                } else {
+                    patternCharactersToBeWritten += currentPatternCharacter
+                }
+            }
+            patternIndex = pattern.index(after: patternIndex)
+        }
+    }
+
+    private func handleTextForLettersAndNumbers(currentText: String,
```

Comment:
rename to formatTextForLettersAndNumbers

Reply:
done